src/libraries/icubmod/embObjMotionControl/embObjMotionControl.cpp

```diff
@@ -973,8 +981,33 @@ bool embObjMotionControl::fromConfig_Step2(yarp::os::Searchable &config)
     if(serviceConfig.ethservice.configuration.type == eomn_serv_MC_foc)
     {
         std::string groupName = (static_cast<eObrd_type_t>(serviceConfig.ethservice.configuration.data.mc.foc_based.type) == eobrd_foc) ? "2FOC" : "AMCBLDC";
-        if(!_mcparser->parseFocGroup(config, _foc_based_info, groupName))
+        if(!_mcparser->parseFocGroup(config, _foc_based_info, groupName, _temperatureSensorsVector))
             return false;
+
+        for (j = 0; j < _njoints; j++)
+        {
+            if (((_foc_based_info[j].temperatureSensorType != motor_temperature_sensor_none ) || (_foc_based_info[j].hasTempSensor != 0)) && ((_temperatureLimits[j].hardwareTemperatureLimit == 0) || (_temperatureLimits[j].warningTemperatureLimit == 0)))
+            {
+                yError() << "In" << getBoardInfo() << "joint" << j << ": inconsistent configuration, please update it. If Temperature limits are not set then TemperatureSensorType must be NONE or not set and/or HasTempSensor must be zero. Aborting...";
+                return false;
+            }
+            
+            //TODO: Add check to _temperatureLimits[j].hardwareTemperatureLimit != 0. If we are here it should be already as this because we are doing check when parsing value from config. 
+            if (_foc_based_info[j].temperatureSensorType == motor_temperature_sensor_pt100)
```

Comment:
you don't need to check the type of the sensor. Automatically the correct function is called based on the type of the object